### DISCUSSION TITLE
community/asset_library/submitting_to_assetlib

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#232] [User-contributed notes]
# community/asset_library/submitting_to_assetlib

Introduction: This tutorial aims to serve as a guide on how you can submit your own assets to the Godot Asset Library and share them with the Godot community. As mentioned in the Using the Asset Li...

https://docs.godotengine.org/en/latest/community/asset_library/submitting_to_assetlib.html

**Comment by walksanatora** [comment]
I cant find any rules. but what are the rules for depending on other plugins

**Reply by Calinou** [reply]
There are no strict rules for depending on other plugins, but it's strongly recommended you do one of these:

- Mention the dependency in your plugin's description *and* README.
- Bundle the other plugin in your plugin if the other plugin's license allows for it. This way, your plugin will work out of the box.

**Comment by chun92** [comment]
I wonder that I can use Autoload in submitting asset library.
Since it is the parts of project setting, I suspect that autoloaded classes won't function correctly.

**Reply by Calinou** [reply]
Editor plugins can be used to [automatically add autoloads when they're enabled](https://docs.godotengine.org/en/stable/tutorials/plugins/editor/making_plugins.html#registering-autoloads-singletons-in-plugins).

For non-editor plugin assets, you will have to ask users to manually add them as autoloads in the Project Settings. This can be done in the asset description and its README file.

**Comment by lixi1983** [comment]
I have provided a GitHub release download link: https://github.com/lixi1983/protoc-gen-gdscript-simple/releases/download/v1.0.0/godot-protobuf-gdscript-plugin-v1.0.0.zip
  But why am I still getting the error: "Error: Using Git tags or branches is no longer supported. Please give a full Git commit hash instead, or use the Custom download provider for GitHub Releases downloads"?

**Reply by Calinou** [reply]
You need to use the Custom download provider instead of GitHub for GitHub Releases links to work correctly.

**Comment by qt911025** [comment]
Where is the config file to specify which the public part is?
In a case that I made a tool in addons, and used some development dependencies. How to mark these dependencies as exception?

**Reply by Calinou** [reply]
By default, the entire repository contents are included in the ZIP archive. If you wish to exclude certain files from the generated ZIP, use the `.gitattributes` file as described on this page.

If you wish to include certain files in the ZIP archive that are not normally present in the repository, you'll need to create ZIP archives that you upload to GitHub Releases and use the **Custom** download provider to specify the download link to the ZIP in question.

**Reply by thiagola92** [reply]
Just to be sure, when you say:

> use the **Custom** download provider to specify the download link to the ZIP in question.

It means setting **Custom** on the **Repository host** field during the Godot Asset form?  
And putting the ZIP link in **Download Commit/URL** field, right?

**Reply by Calinou** [reply]
Yes :slightly_smiling_face:

**Comment by Juanxpeke** [comment]
What happens with the .uid files added in godot 4.4? Could those files cause conflict if they are added to the addon repository?

**Reply by Calinou** [reply]
There should be no conflicts, as your add-on's files should be scoped to `addons/addon_name/` anyway.

**Reply by Juanxpeke** [reply]
So if a user has in its project a resource with UID = X and downloads my addon, and just by coincidence one of the files in my addon has also UID = X, it won't be a problem because the paths are different? (Sorry for asking here, but no one else seems to know :c )

**Reply by Calinou** [reply]
UID collisions should just not happen in practice – while not impossible, the likelihood is *extremely* low. I have yet to see a documented example of an UID collision occurring when merging files from different projects.

**Comment by mikest** [comment]
I keep getting "CSRF check failed." errors after submitting. What am I doing wrong?

**Reply by Calinou** [reply]
This is a known issue, but I've never been able to reproduce it:

- https://github.com/godotengine/godot-asset-library/issues/358

**Reply by mikest** [reply]
I was running into the issue on Edge browser. I didn't run into the issue when I switched to Firefox.

**Comment by dginovker** [comment]
Are proprietary binaries allowed to be submitted?

**Reply by Calinou** [reply]
The current asset library only allows submitting open source assets. If your asset depends on a third-party library that is under a proprietary license, it should be downloaded at runtime by the plugin instead of being included in the source tree.

The upcoming asset store ([currently in beta as of writing](https://store-beta.godotengine.org/)) allows assets under any license, including proprietary licenses.
